Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -690,6 +691,16 @@ open class NavigationMapView: UIView {
         }
         
         let cameraOptions = CameraOptions(cameraState: mapView.cameraState)
+        
+        if let heading = heading {
+            userCourseView.update(location: location,
+                                  pitch: cameraOptions.pitch!,
+                                  direction: heading.trueHeading,
+                                  animated: animated,
+                                  navigationCameraState: navigationCamera.state)
+            return
+        }
+        
         userCourseView.update(location: location,
                               pitch: cameraOptions.pitch!,
                               direction: cameraOptions.bearing!,
```

Comment:
By this point, `cameraOptions.bearing` is already set to something match the course. Where does that happen? Should that code account for heading while walking instead of the conditional above?